### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by JonSteinn**

### CODE REVIEW

File: tests/patterns/test_perm.py [hunk 1]
```diff
@@ -1463,6 +1463,883 @@ def test_major_index():
     assert Perm((7, 0, 4, 3, 1, 5, 2, 6)).major_index() == 14
 
 
+def test_depth():
```
Comment (enadeau) [inline tests/patterns/test_perm.py:1466]: Isn't that a bit too much?

**Review by enadeau — APPROVED** [review]

**Comment by christianbean** [comment]
It would be good to add a small section to the readme, just before the bisc section, to explain how to use this cool new feature.

File: tests/patterns/test_perm.py (hunk 1 above)
Comment (JonSteinn) [reply tests/patterns/test_perm.py:1466]: removed len 6

File: README.rst [hunk 2]
```diff
@@ -200,6 +201,60 @@ given class.
     ...     print(strat.reference())
     Enumeration of Permutation Classes and Weighted Labelled Independent Sets: Corollary 4.3
 
+Permutation statistics
+######################
+
+With the ``PermutationStatistics`` class we can look for distributions of statistics for
```
Comment (christianbean) [inline README.rst:207]: remove s

File: README.rst [hunk 3]
```diff
@@ -200,6 +201,60 @@ given class.
...
+classes and look for statistics preservations (or transformation) either for two classes
+or given a bijection. First we need to import it.
+
+.. code-block:: python
+
+    >>> from permuta.permutils.statistics import PermutationStatistic
+
+To see a distribution for a given statistics we grab its instance and provide a length
```
Comment (christianbean) [inline README.rst:215]: remove s

File: permuta/permutils/statistics.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    ) -> List[List[int]]:
+        """Return a table (i,k) for the distribution of a statistic. Here i=0..n is the
+        length of the permutation and k is the statistic. If a class is not provided,
+        we use the set of all permutations.
+        """
+        return [self.distribution_for_length(i, perm_class) for i in range(n + 1)]
+
+    @classmethod
+    def equally_distributed(cls, class1: Av, class2: Av, n: int = 6) -> Iterator[str]:
+        """Returm all stats that are equally distributed for two classes up to a max
```
Comment (christianbean) [inline permuta/permutils/statistics.py:91]: returm

**Review by christianbean — APPROVED** [review]
Very good! I noticed a couple of typos, but I think it is ready to go!

File: README.rst [hunk 5]
```diff
@@ -200,6 +201,60 @@ given class.
...
+and a class (no class will use the set of all permutations).
+
+.. code-block:: python
+
+    >>> # PermutationStatistic.show_predefined_statistics() # Show all statistics with id
```
Comment (enadeau) [inline README.rst:220]: We should uncomment this line, so people can see what statistic we have

**Pull request merged**